generators/app/index.js

```diff
@@ -170,28 +229,46 @@ module.exports = yeoman.generators.Base.extend({
         this.log('\n' + chalk.bold.green('I\'m Updating selected entities ') + chalk.bold.yellow(this.entitiesToUpdate));
         this.log('\n' + chalk.bold.yellow('Make sure these classes does not extend any other class to avoid any errors during compilation.'));
         var jsonObj = null;
+        this.auditedEntities = [];
+
         this.entitiesToUpdate.forEach(function(entityName) {
-          // extend entity with AbstractAuditingEntity
-          jhipsterFunc.replaceContent(this.javaDir + 'domain/' + entityName + '.java', 'public class ' + entityName, 'public class ' + entityName + ' extends AbstractAuditingEntity');
-          // extend DTO with AbstractAuditingDTO
-          jsonObj = this.fs.readJSON('.jhipster/' + entityName + '.json')
-          if(jsonObj.dto == 'mapstruct') {
-            jhipsterFunc.replaceContent(this.javaDir + 'web/rest/dto/' + entityName + 'DTO.java', 'public class ' + entityName + 'DTO', 'public class ' + entityName + 'DTO extends AbstractAuditingDTO');
-          }
+          this.auditedEntities.push("\"" + entityName + "\"")
+          if (this.auditFramework === 'custom') {
+            // extend entity with AbstractAuditingEntity
+            jhipsterFunc.replaceContent(this.javaDir + 'domain/' + entityName + '.java', 'public class ' + entityName, 'public class ' + entityName + ' extends AbstractAuditingEntity');
+            // extend DTO with AbstractAuditingDTO
+            jsonObj = this.fs.readJSON('.jhipster/' + entityName + '.json')
+            if(jsonObj.dto == 'mapstruct') {
+              jhipsterFunc.replaceContent(this.javaDir + 'web/rest/dto/' + entityName + 'DTO.java', 'public class ' + entityName + 'DTO', 'public class ' + entityName + 'DTO extends AbstractAuditingDTO');
+            }
+
+            //update liquibase changeset
+            var file = glob.sync(this.resourceDir + "/config/liquibase/changelog/*" + entityName + ".xml")[0];
+            if(file) {
+              var columns = "<column name=\"created_by\" type=\"varchar(50)\">\n" +
+              "                <constraints nullable=\"false\"/>\n" +
+              "            </column>\n" +
+              "            <column name=\"created_date\" type=\"timestamp\" defaultValueDate=\"${now}\">\n" +
+              "                <constraints nullable=\"false\"/>\n" +
+              "            </column>\n" +
+              "            <column name=\"last_modified_by\" type=\"varchar(50)\"/>\n" +
+              "            <column name=\"last_modified_date\" type=\"timestamp\"/>";
+              jhipsterFunc.addColumnToLiquibaseEntityChangeset(file, columns);
+            }
+          } else {
+
+            // check if repositories are already annotated
+            var auditTableAnnotation = '@JaversSpringDataAuditable';
+            var pattern = new RegExp(auditTableAnnotation, 'g')
+            var content = fs.readFileSync(this.javaDir + 'repository/' + entityName + 'Repository.java', 'utf8');
```

Comment:
Fred, the check is fine I was talking about using `this.fs.read` instead of `fs.readFileSync` method as it will do the same and we can avoid importing the fs dependency for this JS file